app/app.go

```diff
@@ -349,6 +366,46 @@ func (a *App) ConfigFileName() string {
 	return utils.CfgFileName
 }
 
+func (a *App) ClientConfig() map[string]string {
+	return a.clientConfig
+}
+
+func (a *App) ClientConfigHash() string {
+	return a.clientConfigHash
+}
+
+func (a *App) regenerateClientConfig() {
+	a.clientConfig = utils.GenerateClientConfig(a.Config(), a.DiagnosticId())
+	clientConfigJSON, _ := json.Marshal(a.clientConfig)
+	a.clientConfigHash = fmt.Sprintf("%x", md5.Sum(clientConfigJSON))
+}
+
+func (a *App) DiagnosticId() string {
+	return a.diagnosticId
+}
+
+func (a *App) SetDiagnosticId(id string) {
+	a.diagnosticId = id
```

Comment:
In a future patch I'll be able to eliminate some ancient race conditions by making all of these functions safe for concurrent use via mutexes or atomics.